src/controllers/things_controller.js

```diff
@@ -490,26 +551,39 @@ ThingsController.ws('/:thingId/', (websocket, request) => {
     try {
       request = JSON.parse(requestText);
     } catch (e) {
-      sendMessage(JSON.stringify({
+      sendMessage({
         messageType: Constants.ERROR,
         data: {
           status: '400 Bad Request',
           message: 'Parsing request failed',
         },
-      }));
+      });
+      return;
+    }
+
+    const id = request.id || thingId;
```

Comment:
Line 620 below should probably use `id`, rather than `thingId`.

Reply:
Nice catch, thanks!